File: Java/src/test/arrays/TwoSumTest.java
package test.arrays;

import arrays.TwoSum;
import org.junit.Assert;
import org.junit.Test;
import org.junit.Before;
import org.junit.After;

/**
 * TwoSum Tester.
 *
 * @author IrvingRyan
 * @version 1.0
 * @since <pre>3月 27, 2020</pre> Í
 */
public class TwoSumTest {

    private int[] arr;

    @Before
    public void before() throws Exception {
        arr = new int[]{2, 7, 11, 15, 20, 30, 44, 65, 54};
    }

    @Test
    public void testTwoSum() {
        TwoSum twoSum = new TwoSum();
        int[] result1 = twoSum.twoSum(arr, 10);
        Assert.assertArrayEquals(null, result1);

        int[] result2 = twoSum.twoSum(arr, 50);
        Assert.assertArrayEquals(new int[]{4, 5}, result2);

        int[] result3 = twoSum.twoSum(arr, 51);
        Assert.assertArrayEquals(new int[]{1, 6}, result3);

    }
    @Test
    public void testTwoSum2() {
        TwoSum twoSum = new TwoSum();
        int[] result1 = twoSum.twoSum2(arr, 10);
        Assert.assertArrayEquals(null, result1);

        int[] result2 = twoSum.twoSum2(arr, 50);
        Assert.assertArrayEquals(new int[]{4, 5}, result2);

        int[] result3 = twoSum.twoSum2(arr, 51);
        Assert.assertArrayEquals(new int[]{1, 6}, result3);

    }

    @After
    public void after() throws Exception {
        arr = null;
    }


} 

File: Kotlin/src/test/array/TwoSumTest.kt
package test.array

import arrays.TwoSum
import org.junit.After
import org.junit.Assert.assertArrayEquals
import org.junit.Before
import org.junit.Test

class TwoSumTest {
    private var arr: IntArray? = null

    @Before
    @Throws(Exception::class)
    fun before() {
        arr = intArrayOf(2, 7, 11, 15, 20, 30, 44, 65, 54)
    }

    @Test
    fun testTwoSum() {
        val twoSum = TwoSum()
        val result1 = twoSum.twoSum(arr!!, 10)
        assertArrayEquals(null, result1)
        val result2 = twoSum.twoSum(arr!!, 50)
        assertArrayEquals(intArrayOf(4, 5), result2)
        val result3 = twoSum.twoSum(arr!!, 51)
        assertArrayEquals(intArrayOf(1, 6), result3)
    }

    @Test
    fun testTwoSum2() {
        val twoSum = TwoSum()
        val result1 = twoSum.twoSum2(arr!!, 10)
        assertArrayEquals(null, result1)
        val result2 = twoSum.twoSum2(arr!!, 50)
        assertArrayEquals(intArrayOf(4, 5), result2)
        val result3 = twoSum.twoSum2(arr!!, 51)
        assertArrayEquals(intArrayOf(1, 6), result3)
    }

    @After
    @Throws(Exception::class)
    fun after() {
        arr = null
    }

}